base/linalg/uniformscaling.jl

```diff
@@ -178,6 +178,8 @@ function copy!(A::AbstractMatrix, J::UniformScaling)
     return A
 end
 
+cond{T}(J::UniformScaling{T}) = J.λ ≠ zero(T) ? one(T) : convert(T, Inf)
```

Comment:
Since the condition number is dimensionless, this is yet another case of #16116: we need the meaning of `one` as "multiplicative identity", which may not be of the same type as `T`.

By the same token, however, to be type stable the infinite case should be `oftype(one(T), Inf)`.

Or, better yet, `one(J.λ) : oftype(one(J.λ), Inf)`